doc: Expand light switch sample overview doc (#4385)
nrfconnect/sdk-nrf: samples/bluetooth/mesh/light_switch/README.rst

   :local:
   :depth: 2

The :ref:`ug_bt_mesh` light switch sample can be used to change the state of light sources on other devices within the same mesh network.

> It also demonstrates how to use Bluetooth mesh models by using the Generic OnOff model in an application.

Reviewer: ... Generic OnOff client model ...

> Use the light switch sample with the :ref:`bluetooth_mesh_light` sample to demonstrate its function in a basic Bluetooth mesh network.

Reviewer: ```suggestion
Use the light switch sample with the :ref:`bluetooth_mesh_light` sample to demonstrate its function in a Bluetooth mesh network.
```

Otherwise, a reader would wonder what is 'basic' Bluetooth mesh network.

Requirements
************

The sample supports the following development kits:

.. table-from-rows:: /includes/sample_board_rows.txt
   :header: heading
   :rows: nrf5340dk_nrf5340_cpuapp_and_cpuappns, nrf52840dk_nrf52840, nrf52dk_nrf52832, nrf52833dk_nrf52833

You need at least two development kits:

* One development kit where you program this sample application (the client)
* One (or more) development kit(s) where you program the :ref:`bluetooth_mesh_light` sample application (the server(s)), and configure according to the mesh light sample's :ref:`testing guide <bluetooth_mesh_light_testing>`

For provisioning and configuring of the mesh model instances, the sample requires a smartphone with Nordic Semiconductor's nRF Mesh mobile app installed in one of the following versions:

* `nRF Mesh mobile app for Android`_
* `nRF Mesh mobile app for iOS`_

Overview
********

> The Bluetooth mesh light switch sample demonstrates how to set up a basic mesh client model application, and control LEDs with the Bluetooth mesh using the :ref:`bt_mesh_onoff_readme`.

Reviewer 2: Here too "basic" can be removed.

To display any functionality, the sample must be paired with a device with the :ref:`bluetooth_mesh_light` sample running in the same mesh network.

In both samples, devices are nodes with a provisionee role in a mesh network.

> Provisioning is performed in the `nRF Mesh mobile app`_.

Reviewer: ```suggestion
Provisioning is performed with the `nRF Mesh mobile app`_.
```

Author: Actually, looking at it now, I feel neither "in" nor "with" are quite a match. How about
- Provisioning is performed using the ...
or 
- Provisioning is performed from the ...

This mobile application is also used to configure key bindings, and publication and subscription settings of the Bluetooth mesh model instances in the sample to enable them to communicate with the servers.

> The Generic OnOff Client/Server models are used for manipulating the on/off state.

Reviewer: ```suggestion
The Generic OnOff Client model is used for manipulating the Generic OnOff state associated with the Generic OnOff Server model.
```

The light switch sample implements the Generic OnOff Client model.

The sample has four buttons to control the state of LED 1 on servers (implemented by the :ref:`bluetooth_mesh_light` sample).
Four instances of the Generic OnOff Client model are instantiated in the light switch sample, one for each button on the development kit that is used.
When a user presses any of the buttons, an OnOff Set message is sent out to the configured destination address.

> After provisioning and configuring the mesh models supported by the sample in the `nRF Mesh mobile app`_, you can control the LEDs on the development kit from the app.

Reviewer: ```suggestion
After provisioning and configuring the mesh models supported by the sample in the `nRF Mesh mobile app`_, you can control the LEDs on the other (server) development kit(s) from the app.
```

The response from the target device updates the corresponding LED on the mesh light switch device.
If the model is configured to publish to a group address, it calls :c:func:`bt_mesh_onoff_cli_set_unack` instead, to avoid getting responses from multiple devices at once.

.. _bluetooth_mesh_light_switch_user_interface:

User interface
**************

Buttons:

>       During the provisioning process, the buttons (1 to 4) can be used for OOB input.

Reviewer 2: Is this correct? It looks the the button handler returns before it is provisioned.
```suggestion
      During the provisioning process, the buttons (1 to 4) is not used.
```

Author: Here I just had a sentence from Trond to lean on (når provisioning foregår, kan knappene brukes til "OOB input"). @trond-snekvik can you please confirm that it is ok for you too if I change to Ingar's suggestion :)

Reviewer 3: @mia-ko's text is correct, provisioning has its own button handler here:
https://github.com/nrfconnect/sdk-nrf/blob/master/subsys/bluetooth/mesh/dk_prov.c#L75-L87

.. note::
   The light switch sample cannot demonstrate any functionality on its own, and needs a device with the :ref:`bluetooth_mesh_light` sample running in the same mesh network.
   Before testing mesh light switch, go through the mesh light's :ref:`testing guide <bluetooth_mesh_light_testing>` with a different development kit.

> After programming the sample to your development kit, you can test it by using a smartphone with Nordic Semiconductor's nRF Mesh app installed.

Reviewer 2: ```suggestion
After programming the sample to your development kit, you can test it by using a smartphone with `nRF Mesh mobile app`_ installed.
```
